File: Byte&Buy.Backend/src/ByteBuy.Core/DTO/Public/Offer/RentOffer/RentOfferDetailsResponse.cs
using ByteBuy.Core.DTO.Public.Image;
using ByteBuy.Core.DTO.Public.Money;

namespace ByteBuy.Core.DTO.Public.Offer.RentOffer;

public record RentOfferDetailsResponse(
    Guid Id,
    int MaxRentalDays,
    int QuantityAvaliable,
    MoneyDto PricePerDay,
    string Condition,
    string Category,
    string Description,
    string Title,
    Object Seller,
    IReadOnlyCollection<ImageResponse> Images);

File: Byte&Buy.AngularSPA/src/app/core/dto/offers/rent/rent-offer-details-response.ts
import { Guid } from "guid-typescript";
import { MoneyDto } from "../../common/money-dto";
import { CompanySellerResponse } from "../common/company-seller-response";
import { PrivateSellerResponse } from "../common/private-seller-response";
import { ImageResponse } from "../../image/image-response";
import { OfferStatus } from "../enum/offer-status";

export interface RentOfferDetailsResponse {
  id: Guid;
  maxRentalDays: number;
  status: OfferStatus;
  quantityAvaliable: number;
  pricePerDay: MoneyDto;
  condition: string;
  category: string;
  description: string;
  title: string;
  seller: CompanySellerResponse | PrivateSellerResponse;
  images: ImageResponse[];
}
